core/src/main/java/de/jplag/options/JPlagOptions.java

```diff
@@ -225,6 +258,46 @@ private Integer normalizeMinimumTokenMatch(Integer minimumTokenMatch) {
         return (minimumTokenMatch != null && minimumTokenMatch < 1) ? Integer.valueOf(1) : minimumTokenMatch;
     }
 
+    public interface PreParseHook {
+        void callback(List<Submission> submissions);
+    }
+
+    public static final PreParseHook DEFAULT_PRE_PARSE_HOOK = new PreParseHook() {
```

Comment:
I'd suggest to use (submissionList) -> log.debug("PreParse {} submissions", submissionList.size()) or something similar as default

Reply:
I haven't done this, because we only have access to the `JPlagOptions` logger and so every line will start with `JPlagOptions` instead of the class where the hook was actually called from, and passing the `logger` to use into these hooks seems a bit clunky.